### PR TITLE
Wallet fix unstaking bug

### PR DESCRIPTION
**Pull request by Jibz1**
- Removed the form component from unstake 
- added Available SUI balance
- Fixed unstaking bug requiring Amount field
- set rewards to absolute value. 
- fixed typo 
Screenshot 2023-01-26 at 5 17 31 PM

### CODE REVIEW

File: apps/wallet/src/ui/app/staking/stake/StakeForm.tsx [hunk 1]
```diff
@@ -56,9 +61,28 @@ function StakeForm({
         setFieldValue('amount', maxToken);
     }, [maxToken, setFieldValue]);
 
+    const calculateRemaining = useMemo(() => {
+        if (!maxToken) return 0;
+        return roundFloat(+maxToken - +amount, DECIMAL_PLACE);
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/staking/stake/StakeForm.tsx:66]: We should do this calculation with mist/not dominated values and use `useFormatCoin` to do the final format to display it

File: apps/wallet/src/ui/app/staking/getStakingRewards.ts [hunk 2]
```diff
@@ -42,5 +42,5 @@ export function getStakingRewards(
         .dividedBy(delegationTokenSupply);
 
     const earnToken = currentSuiWorth.decimalPlaces(0, 1).minus(pricipalAmout);
-    return earnToken.toNumber();
+    return Math.abs(earnToken.toNumber());
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/staking/getStakingRewards.ts:45]: Why abs is necessary. Earned should always be > 0 right?

File: apps/wallet/src/ui/app/staking/stake/StakeForm.tsx (hunk 1 above)
Comment (Jibz1) [reply apps/wallet/src/ui/app/staking/stake/StakeForm.tsx:66]: Got it

File: apps/wallet/src/ui/app/staking/stake/StakeForm.tsx [hunk 3]
```diff
@@ -21,30 +22,45 @@ export type StakeFromProps = {
...
 }: StakeFromProps) {
-    const { setFieldValue } = useFormikContext<FormValues>();
+    const {
+        setFieldValue,
+        values: { amount },
+    } = useFormikContext<FormValues>();
 
     const onClearRef = useRef(onClearSubmitError);
     onClearRef.current = onClearSubmitError;
+    const gasBudget = BigInt(
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/staking/stake/StakeForm.tsx:54]: Lets remove this as it is confusing. Gas budget is always the DEFAULT_GAS_BUDGET_FOR_STAKE.

File: apps/wallet/src/ui/app/staking/stake/StakeForm.tsx [hunk 4]
```diff
@@ -21,30 +22,45 @@ export type StakeFromProps = {
...
 
     const [gasBudgetEstimation] = useFormatCoin(
         DEFAULT_GAS_BUDGET_FOR_STAKE,
         SUI_TYPE_ARG
     );
 
-    const coinBalanceMinusGas =
-        coinBalance -
-        BigInt(coinType === SUI_TYPE_ARG ? DEFAULT_GAS_BUDGET_FOR_STAKE : 0);
+    const coinBalanceMinusGas = coinBalance - gasBudget;
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/staking/stake/StakeForm.tsx:63]: we can do something like
```suggestion
    const totalAvailableBalance = coinBalance - BigInt(coinType === SUI_TYPE_ARG ? DEFAULT_GAS_BUDGET_FOR_STAKE : 0);
```

File: apps/wallet/src/ui/app/staking/stake/StakeForm.tsx [hunk 5]
```diff
@@ -21,30 +22,45 @@ export type StakeFromProps = {
     submitError: string | null;
     coinBalance: bigint;
     coinType: string;
+    coinDecimals: number;
     epoch: string;
     onClearSubmitError: () => void;
 };
 
+function AvailableBalance({ amount }: { amount: bigint }) {
+    const [formatted, symbol] = useFormatCoin(amount, SUI_TYPE_ARG);
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/staking/stake/StakeForm.tsx:31]: How do you know that amount is in SUI? Given that `StakeForm` accepts `coinType` to know the type of the coin either we need to use that here as well or remove that requirement from everywhere if we know that we only are going to support staking for SUI.

File: apps/wallet/src/ui/app/staking/stake/StakeForm.tsx [hunk 6]
```diff
@@ -56,9 +72,21 @@ function StakeForm({
         setFieldValue('amount', maxToken);
     }, [maxToken, setFieldValue]);
 
+    const calculateRemaining = useMemo(() => {
+        if (!coinBalance) return 0n;
+        const bigIntAmount = parseAmount(amount, coinDecimals);
+        return coinBalance - bigIntAmount - gasBudget;
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/staking/stake/StakeForm.tsx:78]: Just use 
```suggestion
        return totalAvailableBalance - bigIntAmount;
```

Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/staking/stake/StakeForm.tsx:78]: Also about this do we really want to change the available balance based on the amount that the user entered to stake? In other places eg send tokens we show always the total balance of the user.
cc @mystie711

**Comment by mystie711** [comment]
The spacing looks different from figma
image

vs code
image

**Comment by Jibz1** [comment]
> The spacing looks different from figma as does color of the text, image
> 
> vs code image

Was it alway 12px, I will update the spacing, however, the color is correct

**Comment by mystie711** [comment]
> > The spacing looks different from figma as does color of the text, image
> > vs code image
> 
> Was it alway 12px, I will update the spacing, however, the color is correct ` <Text variant="bodySmall" color="steel" weight="medium">` Steel

I still think that the second line color is darker than "Sui Steel" - the color speced in Figma. Is there a wallet build i could install to check the styles in my own? 🤔

**Pull request merged**